### PR TITLE
Removing pinot-common dependency from pinot-hadoop-filesystem and pin…

### PR DESCRIPTION
**Pull request by kishoreg**
…ot-azure-filesystem

Prior to this PR, pinot-hadoop-filesystem and pinot-azure-filesystem implementations were responsible for retrying. This PR moves the responsibility to caller(pinot). This also allowed us to remove the dependency on pinot-common.

With this PR, all file-system implementations are decoupled from pinot-core and pinot-common.
pinot-record-readers were already decoupled from pinot-core and pinot-common. 
Last thing that needs to be decoupled in pinot-kafka-connectors

### CODE REVIEW

File: pinot-azure-filesystem/src/main/java/org/apache/pinot/filesystem/AzurePinotFS.java [hunk 1]
```diff
@@ -56,6 +55,10 @@
   private static final int BUFFER_SIZE = 4096;
   private ADLStoreClient _adlStoreClient;
   private static final String[] EMPTY_ARR = new String[0];
+  private static final String ACCOUNT_ID = "accountId";
```
Comment (Jackie-Jiang) [inline pinot-azure-filesystem/src/main/java/org/apache/pinot/filesystem/AzurePinotFS.java]: Better to make them public?

File: pinot-common/src/main/java/org/apache/pinot/filesystem/PinotFSDelegator.java [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+import java.io.InputStream;
+import java.net.URI;
+import org.apache.commons.configuration.Configuration;
+import org.apache.pinot.common.utils.retry.RetryPolicies;
+import org.apache.pinot.common.utils.retry.RetryPolicy;
+import org.apache.pinot.spi.filesystem.PinotFS;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static org.apache.pinot.common.utils.CommonConstants.SegmentOperations.RETRY;
```
Comment (Jackie-Jiang) [inline pinot-common/src/main/java/org/apache/pinot/filesystem/PinotFSDelegator.java]: Please avoid static import for code convention

File: pinot-common/src/main/java/org/apache/pinot/filesystem/PinotFSDelegator.java [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+  @Override
+  public String[] listFiles(URI fileUri, boolean recursive)
+      throws IOException {
+    return _pinotFS.listFiles(fileUri, recursive);
+  }
+
+  @Override
+  public void copyToLocalFile(URI srcUri, File dstFile)
+      throws Exception {
+    RetryPolicy fixedDelayRetryPolicy = RetryPolicies.fixedDelayRetryPolicy(_retryCount, _retryWaitMs);
```
Comment (Jackie-Jiang) [inline pinot-common/src/main/java/org/apache/pinot/filesystem/PinotFSDelegator.java]: I think the main purpose for this class is to use the RetryPolicy. I found the same issue for other classes (e.g. fetching schema). Maybe it is better to move the RetryPolicy into pinot-spi as util classes?

File: pinot-hadoop-filesystem/src/main/java/org/apache/pinot/filesystem/HdfsSegmentFetcher.java [hunk 4]
```diff
@@ -1,135 +0,0 @@
...
-import static org.apache.pinot.common.utils.CommonConstants.SegmentOperations.HadoopSegmentOperations.PRINCIPAL;
-import static org.apache.pinot.common.utils.CommonConstants.SegmentOperations.RETRY;
-import static org.apache.pinot.common.utils.CommonConstants.SegmentOperations.RETRY_DEFAULT;
-import static org.apache.pinot.common.utils.CommonConstants.SegmentOperations.RETRY_WAITIME_MS;
-import static org.apache.pinot.common.utils.CommonConstants.SegmentOperations.RETRY_WAITIME_MS_DEFAULT;
-
-
-// Use PinotFSSegmentFetcher instead
-@Deprecated
-public class HdfsSegmentFetcher implements SegmentFetcher {
```
Comment (Jackie-Jiang) [inline pinot-hadoop-filesystem/src/main/java/org/apache/pinot/filesystem/HdfsSegmentFetcher.java]: You might need to check with @jamesyfshao before removing it

File: pinot-common/src/main/java/org/apache/pinot/filesystem/PinotFSDelegator.java (hunk 3 above)
Comment (kishoreg) [reply pinot-common/src/main/java/org/apache/pinot/filesystem/PinotFSDelegator.java]: I can do that. With that I can move PinotFsFactory and other classes related to PinotFS to pinot-spi. we need to create a utils module at some point.

File: pinot-azure-filesystem/src/main/java/org/apache/pinot/filesystem/AzurePinotFS.java (hunk 1 above)
Comment (kishoreg) [reply pinot-azure-filesystem/src/main/java/org/apache/pinot/filesystem/AzurePinotFS.java]: Done!

File: pinot-common/src/main/java/org/apache/pinot/filesystem/PinotFSDelegator.java (hunk 2 above)
Comment (kishoreg) [reply pinot-common/src/main/java/org/apache/pinot/filesystem/PinotFSDelegator.java]: Fixed it.

**Comment by mayankshriv** [comment]
@jenniferdai Do the existing tests provide enough coverage for PinotFs, to test these changes?

**Pull request merged**